driver-core/src/main/com/mongodb/internal/connection/SaslAuthenticator.java

```diff
@@ -121,6 +126,16 @@ private void throwIfSaslClientIsNull(final SaslClient saslClient) {
         }
     }
 
+    private void verifySaslClientComplete(final SaslClient saslClient, final SingleResultCallback<Void> callback) {
+        if (saslClient.isComplete()) {
+            callback.onResult(null, null);
+        } else {
+            throw new MongoSecurityException(getMongoCredential(),
```

Comment:
This should be 

     callback.onResult(null, new MongoSecurityException(...))

The fact that this didn't cause a test failure implies a missing test...

Reply:
Done.